### ISSUE TITLE
aspnet-codegenerator razorpage  -outDir Pages/Movie -> namespace is Pages_Movie

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Scaffolding#582]
See [Scaffold the Movie model](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model#scaffold)

When you run the following command:
`dotnet aspnet-codegenerator razorpage -m Movie -dc MovieContext -udl -outDir Pages/Movie`

The *Pages/Movie* `PageModel` classes have the following namespace:

`namespace <Project Name>.Pages_Movie`
The namespace should follow convention and use `Pages.Movie`

**Comment by Rick-Anderson** [comment]
@prafullbhosale

**Comment by prafullbhosale** [comment]
@Rick-Anderson , if I try with `dotnet aspnet-codegenerator razorpage -m Movie -dc MovieContext -udl -outDir Pages\Movie`

i.e. `\` instead of `/`, it generates the namespace properly. 

Looks like a bug in determining namespaces that it doesn't handle `/`.

**Comment by Rick-Anderson** [comment]
Do we need to open another bug for the fact that when it generates namespaces with underscore in lieu of dot (that is `namespace <Project Name>.Pages_Movie`), changing to dot notation causes a strange error that it can't resolve some `#using` statements. I demo'd this to Dan.Roth - do you want a quick demo?

**Comment by prafullbhosale** [comment]
Did you change the namespace in both the `.cshtml` and the `.cshtml.cs` (PageModel class ) files? 
It'd be great to get a quick demo.

**Comment by Rick-Anderson** [comment]
Yes,, although changing in the Razor file wouldn't impact the bug in the .cs file.

**Comment by Rick-Anderson** [comment]
namespace conflict solved - I was using movies as a type an namespace. The _ masked my ambiguity.

**Comment by Rick-Anderson** [comment]
@prafullbhosale  on MacOS, I can't use the work-around, it appends the folder name to the page name. I'm  forced to have 3 versions of this command:

* Windows 
* MacOS
* VS Code (which can be on Windows or anything else)

**Comment by Rick-Anderson** [comment]
@prafullbhosale can we get this fixed for 3.0? It looks bad to have different CLI commands for each OS and is a pain point to document.

**Comment by vijayrkn** [comment]
Tagging @deepchoudhery to look at this for the 5.0 timeframe.

**Comment by Rick-Anderson** [comment]
@deepchoudhery  any chance this will be fixed for 5.0 RTW?

**Comment by Rick-Anderson** [comment]
@deepchoudhery can you confirm this is fixed?   see  https://github.com/dotnet/AspNetCore.Docs/issues/21918
I was hoping to be notified when this was fixed cc: @danroth27 @amarie401

**Comment by Rick-Anderson** [comment]
@vijayrkn this bug is still a PITA for .NET 7 RC2
Can we get this fixed for .NET 8?
cc @tdykstra
